### PR TITLE
fix(webpack): enable SourceMaps in releases

### PR DESCRIPTION
**Pull request by glowcloud**
Refs #9101 

From the official Webpack documentation for [devtool](https://webpack.js.org/configuration/devtool/#production):

Screenshot 2024-04-26 at 11 16 27

and guide for [production](https://webpack.js.org/guides/production/#source-mapping):

Screenshot 2024-04-26 at 11 18 44

Testing the options with the static HTML page in the `dist` folder:
- `source-map` (PR):
source-map

- `nosources-source-map` (current):
nosources-source-map

- `hidden-source-map`:
hidden-source-map

After building with the `source-map` option, the size of the `dist` folder went from `10 439 023 bytes (10,5 MB on disk)` to `19 239 017 bytes (19,3 MB on disk) `

### CODE REVIEW

**Comment by char0n** [comment]
**TODO**:

- we need to limit source map generation to certain build fragments (exports/browser only)
- we need to return `sourceMap` setting for Terser plugin
- we need to refactor the logic of determining if source map is to be generated 
- we need to test on Safari
- ideally close all the source map related issues as resolved

We decided that only certain build fragments that can/may be used in development env will contain source maps. Other will not as we're assuming nobody debugs in production and on production bundles (debugging in mostly done in development).

File: webpack/_config-builder.js [hunk 1]
```diff
@@ -133,7 +133,7 @@ function buildConfig(
       // Otherwise, provide heavy souremaps suitable for development
       devtool: sourcemaps
```
Comment (char0n) [inline webpack/_config-builder.js:134]:
```js
devtool: sourcemaps && minimize ? "nosources-source-map" : "cheap-module-source-map",
```

Comment (char0n) [inline webpack/_config-builder.js:134]:
```js
sourcemaps && minimize
 ? "source-map"
 : sourcemaps && !minimize
 ? "cheap-module-source-map"
 : false
```

**Comment by glowcloud** [comment]
`dist` size after building: `11 804 809 bytes (11,9 MB on disk)`

Tested with a React application in Chrome: 

Screenshot 2024-04-29 at 15 27 20

Firefox:

Screenshot 2024-04-29 at 15 28 39

 and Safari:

Screenshot 2024-04-29 at 15 30 54

**Comment by glowcloud** [comment]
There's also this line in es-bundle-core:
https://github.com/swagger-api/swagger-ui/blob/b6805e5635557f45b255d35b32395c8e4770beef/webpack/es-bundle-core.js#L47
perhaps it was to bypass the previous `nosources-source-map` setting?

**Comment by char0n** [comment]
We don't want to generate source maps for `standalone` build UMD fragment because of:

> We decided that only certain build fragments that can/may be used in development env will contain source maps. Other will not as we're assuming nobody debugs in production and on production bundles (debugging in mostly done in development).

**Comment by char0n** [comment]
> There's also this line in es-bundle-core
> perhaps it was to bypass the previous `nosources-source-map` setting?

We'll use new reusable helper for this one to align it with `swagger-ui.js` build fragment. This one always worked and generated properly working source map, but now we utilize some optimization on top of it.

**Comment by char0n** [comment]
Please also modify https://github.com/swagger-api/swagger-ui/blob/955c0ddbac8674aeaa99c9fba714248f4c2639ad/flavors/swagger-ui-react/release/run.sh#L12

**Review by char0n — APPROVED** [review]

**Comment by char0n** [comment]
Another references in https://github.com/swagger-api/swagger-ui/blob/78f6ffb5e5d307adbe2ec9d564be7a4b3511f877/.npmignore#L8

**Pull request merged**